### PR TITLE
Modification of is_constant to take into account the MatrixSymbol case.

### LINKED ISSUE #19162: MatrixSymbol inversion fails.
The code:

```python
X = sp.Matrix(sp.MatrixSymbol('X', 2, 2))
X.inv()
```

generates the error:

```
TypeError: 'Zero' object is not subscriptable
```

### PR DESCRIPTION
**Pull request by jmgc**
Fixes #19162

#### Brief description of what is fixed or changed

The present issue was detected when calculating the inverse of a MatrixSymbol. The reason came from the is_constant method that did not take into account the case of MatrixSymbol giving the error that the zero value is not subscriptable.

#### Other comments

A test has been added to test_matrices to check this case.

#### Release Notes

* matrices
    * Solved a bug that prevented the use of the MatrixSymbol inversion.

### CODE REVIEW

**Comment by sylee957** [comment]
Does this work when it reaches random expression substitution cases like `expr._random(None, 1, 0, 1, 0)`?

**Comment by jmgc** [comment]
It passes the different tests, however, I do not know if it reaches this specific line.

File: sympy/core/expr.py [hunk 1]
```diff
@@ -601,6 +601,7 @@ def is_constant(self, *wrt, **flags):
         >>> ((one - 1)**(x + 1)).is_constant() in (True, False) # could be 0 or 1
         True
         """
+        from sympy.matrices import ZeroMatrix, ones
```
Comment (oscarbenjamin) [inline sympy/core/expr.py:604]: We shouldn't introduce dependency on matrices into the core.

**Comment by oscarbenjamin** [comment]
I think that this is not really the right approach to fixing the issue. I suggested something in https://github.com/sympy/sympy/issues/19162#issuecomment-620631559 that `MatrixElement` should check its arguments in the constructor.

**Comment by czgdp1807** [comment]
> I suggested something in #19162 (comment) that MatrixElement should check its arguments in the constructor.

@jmgc Any thoughts on this? Please let us know if you are proceeding with this PR. Thanks for your contributions. 

Please do not close this PR.

**Comment by jmgc** [comment]
My understanding is that if the solution proposed by @oscarbenjamin is more general, I think it is the one that should be used. However, the tests included in the present PR shall be kept to avoid new unexpected behaviour.

Thanks

**Comment by czgdp1807** [comment]
Closing this PR in favor of https://github.com/sympy/sympy/pull/20691

**Pull request closed without merging**